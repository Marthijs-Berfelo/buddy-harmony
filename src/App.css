
.fretboard {
  text-anchor: middle;
}

.fretboard-string {
  stroke: #2e2e2e;
}

.fretboard-fret {
  stroke: #706e6e;
}

.fretboard-nut {
  stroke: #2e2e2e;
}

.fretboard-fret-numbers {}

.fretboard-fret-number {
  font-family: sans-serif;
  stroke-width: 1;
  font-size: 24px;
}
