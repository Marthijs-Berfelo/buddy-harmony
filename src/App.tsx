import React from 'react';
import logo from './logo.svg';
import './App.css';
import * as gs from 'guitar-scales';
import guitar from 'react-fretboard-diagram';
import { Tuning } from 'fretboard-api';
const { Diagram } = guitar;

function App() {
  const guitar = gs.GuitarScale;
  const scale = guitar.get('a', 'major blues');
  const cMajor = ['2 4, 1 2 4, 1 3 4, 1 3 4, 2 4, 1 2 4'];
  console.log('SCALES', guitar.getNames());
  console.log('SCALE', scale);
  // printConsoleLog(scale);
  return (
    <div>
      <header className="App-header">
        <img src={logo} className="App-logo" alt="logo" />
        <p>
          Edit <code>src/App.tsx</code> and save to reload.
        </p>
        <a
          className="App-link"
          href="https://reactjs.org"
          target="_blank"
          rel="noopener noreferrer"
        >
          Learn React
        </a>
      </header>
      <div style={{ height: '0.5vh', marginBottom: 32 }}>
        <Diagram
          mouseMoveHandler={undefined}
          className={''}
          mouseClickHandler={undefined}
          frets={12}
          orientation={'vertical'}
          tuning={Tuning.guitar.standard}
          shapes={cMajor}
          fretNumbers={'roman'}
          text={'note'}
          fretNumbersPosition={'left'}
          debug={true}
          diagramStyle={{
            paddingLeft: 200,
            paddingRight: 50,
            paddingTop: 100,
            paddingBottom: 100,
            stringInterval: 200,
            stringWidth: 20,
            fretInterval: 300,
            fretWidth: 30,
            dotIn: 150,
            dotOut: 100,
            dotRadius: 60,
            dotStroke: 10,
            fontSize: 50,
          }}
        />
      </div>
    </div>
  );
}

export default App;
