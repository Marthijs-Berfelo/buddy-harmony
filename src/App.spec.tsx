import React from 'react';
import { render } from '@testing-library/react';
import App from './App';

test('renders app', async () => {
  const { container, findAllByText } = render(<App />);
  await findAllByText('title');
  expect(container).toMatchSnapshot();
});
