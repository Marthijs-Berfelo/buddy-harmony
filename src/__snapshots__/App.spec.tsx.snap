// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders app 1`] = `
<div>
  <div
    class="flex-col justify-center align-top"
    id="app"
  >
    <nav
      class="flex flex-nowrap justify-start w-full max-w-screen-2xl rounded-xl py-4 px-8 shadow-md backdrop-saturate-200 backdrop-blur-2xl bg-opacity-80 border border-white/80 w-full max-w-full rounded-none px-4 bg-white text-white flex flex-row bg-green-100 border-green-100 z-50 fixed"
    >
      <div
        class="flex flex-grow justify-between items-center text-green-900"
      >
        <p
          class="block antialiased font-sans text-base font-light leading-relaxed bg-clip-text text-transparent bg-gradient-to-tr from-green-600 to-green-400 py-1.5 mr-4 font-sans font-bold text-green-800 text-2xl"
        >
          title
        </p>
        <ul
          class="flex items-center gap-6"
        >
          <li
            class="block antialiased font-sans text-xl font-normal leading-relaxed text-inherit p-1 font-normal"
          >
            <a
              class=""
              href="/chord"
            >
              <button
                class="middle none font-sans font-bold center uppercase transition-all text-xs py-3 px-6 rounded-lg border border-blue-500 text-blue-500 hover:opacity-75 focus:ring focus:ring-blue-200 active:opacity-[0.85] flex items-center border-green-600 text-green-600"
                type="button"
              >
                routing.page
              </button>
            </a>
          </li>
          <li
            class="block antialiased font-sans text-xl font-normal leading-relaxed text-inherit p-1 font-normal"
          >
            <a
              class=""
              href="/scale"
            >
              <button
                class="middle none font-sans font-bold center uppercase transition-all text-xs py-3 px-6 rounded-lg border border-blue-500 text-blue-500 hover:opacity-75 focus:ring focus:ring-blue-200 active:opacity-[0.85] flex items-center border-green-600 text-green-600"
                type="button"
              >
                routing.page
              </button>
            </a>
          </li>
          <li
            class="block antialiased font-sans text-xl font-normal leading-relaxed text-inherit p-1 font-normal"
          >
            <a
              class=""
              href="/harmony"
            >
              <button
                class="middle none font-sans font-bold center uppercase transition-all text-xs py-3 px-6 rounded-lg border border-blue-500 text-blue-500 hover:opacity-75 focus:ring focus:ring-blue-200 active:opacity-[0.85] flex items-center border-green-600 text-green-600"
                type="button"
              >
                routing.page
              </button>
            </a>
          </li>
          <li
            class="block antialiased font-sans text-xl font-normal leading-relaxed text-inherit p-1 font-normal"
          >
            <a
              class=""
              href="/help"
            >
              <button
                class="middle none font-sans font-bold center uppercase transition-all text-xs py-3 px-6 rounded-lg border border-blue-500 text-blue-500 hover:opacity-75 focus:ring focus:ring-blue-200 active:opacity-[0.85] flex items-center border-green-600 text-green-600"
                type="button"
              >
                routing.page
              </button>
            </a>
          </li>
        </ul>
        <div
          class="flex"
          id="lang-selector"
        >
          <div
            class="ReactFlagsSelect-module_flagsSelect__2pfa2"
            data-testid="rfs"
          >
            <button
              aria-expanded="false"
              aria-haspopup="listbox"
              aria-labelledby="rfs-btn"
              class="ReactFlagsSelect-module_selectBtn__19wW7"
              data-testid="rfs-btn"
              id="rfs-btn"
              style="font-size: 16px;"
              type="button"
            >
              <span
                class="ReactFlagsSelect-module_selectValue__152eS"
              >
                Select a country
              </span>
            </button>
          </div>
        </div>
      </div>
    </nav>
    <div
      class="flex justify-center pt-24"
      id="content"
      style="width: 100vw;"
    />
    <nav
      class="flex flex-nowrap justify-start w-full max-w-screen-2xl rounded-xl py-4 px-8 shadow-md backdrop-saturate-200 backdrop-blur-2xl bg-opacity-80 border border-white/80 w-full max-w-full rounded-none px-4 bg-white text-white flex flex-row bg-green-100 border-green-100 fixed bottom-0"
    >
      <div
        class="flex flex-grow justify-between items-center text-green-900"
      >
        <div
          class="flex flex-col"
        >
          <a
            class="block antialiased font-sans text-base font-light leading-relaxed text-inherit font-sans text-sm"
            href="https://github.com/Marthijs-Berfelo/buddy-harmony/releases/tag/0.0.16"
            rel="noreferrer"
            target="_blank"
          >
            app-version
          </a>
          <div
            class="flex flex-row items-center justify-evenly mt-1"
          >
            <a
              class="block antialiased font-sans text-base font-light leading-relaxed text-inherit font-sans text-sm flex"
              href="https://github.com/Marthijs-Berfelo/buddy-harmony"
              rel="noreferrer"
              target="_blank"
            >
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-github flex text-xl text-black mr-2"
                data-icon="github"
                data-prefix="fab"
                focusable="false"
                role="img"
                viewBox="0 0 496 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M165.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3 .3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6zm-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5 .3-6.2 2.3zm44.2-1.7c-2.9 .7-4.9 2.6-4.6 4.9 .3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9zM244.8 8C106.1 8 0 113.3 0 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C428.2 457.8 496 362.9 496 252 496 113.3 383.5 8 244.8 8zM97.2 352.9c-1.3 1-1 3.3 .7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1zm-10.8-8.1c-.7 1.3 .3 2.9 2.3 3.9 1.6 1 3.6 .7 4.3-.7 .7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3 .7zm32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3 .7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1zm-11.4-14.7c-1.6 1-1.6 3.6 0 5.9 1.6 2.3 4.3 3.3 5.6 2.3 1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2z"
                  fill="currentColor"
                />
              </svg>
            </a>
            <a
              class="block antialiased font-sans text-base font-light leading-relaxed text-inherit font-sans text-sm flex"
              href="https://github.com/Marthijs-Berfelo/buddy-harmony/issues"
              rel="noreferrer"
              target="_blank"
            >
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-bug flex text-xl text-black mr-2"
                data-icon="bug"
                data-prefix="fas"
                focusable="false"
                role="img"
                viewBox="0 0 512 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M352 96V99.56C352 115.3 339.3 128 323.6 128H188.4C172.7 128 159.1 115.3 159.1 99.56V96C159.1 42.98 202.1 0 255.1 0C309 0 352 42.98 352 96zM41.37 105.4C53.87 92.88 74.13 92.88 86.63 105.4L150.6 169.4C151.3 170 151.9 170.7 152.5 171.4C166.8 164.1 182.9 160 199.1 160H312C329.1 160 345.2 164.1 359.5 171.4C360.1 170.7 360.7 170 361.4 169.4L425.4 105.4C437.9 92.88 458.1 92.88 470.6 105.4C483.1 117.9 483.1 138.1 470.6 150.6L406.6 214.6C405.1 215.3 405.3 215.9 404.6 216.5C410.7 228.5 414.6 241.9 415.7 256H480C497.7 256 512 270.3 512 288C512 305.7 497.7 320 480 320H416C416 344.6 410.5 367.8 400.6 388.6C402.7 389.9 404.8 391.5 406.6 393.4L470.6 457.4C483.1 469.9 483.1 490.1 470.6 502.6C458.1 515.1 437.9 515.1 425.4 502.6L362.3 439.6C337.8 461.4 306.5 475.8 272 479.2V240C272 231.2 264.8 224 255.1 224C247.2 224 239.1 231.2 239.1 240V479.2C205.5 475.8 174.2 461.4 149.7 439.6L86.63 502.6C74.13 515.1 53.87 515.1 41.37 502.6C28.88 490.1 28.88 469.9 41.37 457.4L105.4 393.4C107.2 391.5 109.3 389.9 111.4 388.6C101.5 367.8 96 344.6 96 320H32C14.33 320 0 305.7 0 288C0 270.3 14.33 256 32 256H96.3C97.38 241.9 101.3 228.5 107.4 216.5C106.7 215.9 106 215.3 105.4 214.6L41.37 150.6C28.88 138.1 28.88 117.9 41.37 105.4H41.37z"
                  fill="currentColor"
                />
              </svg>
            </a>
          </div>
        </div>
        <div
          class="flex flex-col"
        >
          <p
            class="block antialiased font-sans text-base font-light leading-relaxed text-inherit font-sans text-sm"
          >
            app-creator
          </p>
          <div
            class="flex flex-row items-center justify-evenly mt-1"
          >
            <a
              class="block antialiased font-sans text-base font-light leading-relaxed text-inherit font-sans text-sm flex"
              href="https://github.com/Marthijs-Berfelo"
              rel="noreferrer"
              target="_blank"
            >
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-github flex text-xl text-black mr-2"
                data-icon="github"
                data-prefix="fab"
                focusable="false"
                role="img"
                viewBox="0 0 496 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M165.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3 .3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6zm-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5 .3-6.2 2.3zm44.2-1.7c-2.9 .7-4.9 2.6-4.6 4.9 .3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9zM244.8 8C106.1 8 0 113.3 0 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C428.2 457.8 496 362.9 496 252 496 113.3 383.5 8 244.8 8zM97.2 352.9c-1.3 1-1 3.3 .7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1zm-10.8-8.1c-.7 1.3 .3 2.9 2.3 3.9 1.6 1 3.6 .7 4.3-.7 .7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3 .7zm32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3 .7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1zm-11.4-14.7c-1.6 1-1.6 3.6 0 5.9 1.6 2.3 4.3 3.3 5.6 2.3 1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2z"
                  fill="currentColor"
                />
              </svg>
            </a>
            <a
              class="block antialiased font-sans text-base font-light leading-relaxed text-inherit font-sans text-sm flex"
              href="https://twitter.com/mwberfelo?t=gh8SH4i2QNAnHqLQ5KyO7A&s=03"
              rel="noreferrer"
              target="_blank"
            >
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-twitter flex text-xl text-black mr-2"
                data-icon="twitter"
                data-prefix="fab"
                focusable="false"
                role="img"
                viewBox="0 0 512 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M459.4 151.7c.325 4.548 .325 9.097 .325 13.65 0 138.7-105.6 298.6-298.6 298.6-59.45 0-114.7-17.22-161.1-47.11 8.447 .974 16.57 1.299 25.34 1.299 49.06 0 94.21-16.57 130.3-44.83-46.13-.975-84.79-31.19-98.11-72.77 6.498 .974 12.99 1.624 19.82 1.624 9.421 0 18.84-1.3 27.61-3.573-48.08-9.747-84.14-51.98-84.14-102.1v-1.299c13.97 7.797 30.21 12.67 47.43 13.32-28.26-18.84-46.78-51.01-46.78-87.39 0-19.49 5.197-37.36 14.29-52.95 51.65 63.67 129.3 105.3 216.4 109.8-1.624-7.797-2.599-15.92-2.599-24.04 0-57.83 46.78-104.9 104.9-104.9 30.21 0 57.5 12.67 76.67 33.14 23.72-4.548 46.46-13.32 66.6-25.34-7.798 24.37-24.37 44.83-46.13 57.83 21.12-2.273 41.58-8.122 60.43-16.24-14.29 20.79-32.16 39.31-52.63 54.25z"
                  fill="currentColor"
                />
              </svg>
            </a>
            <a
              class="block antialiased font-sans text-base font-light leading-relaxed text-inherit font-sans text-sm flex"
              href="mailto:mwberfelo+harmony-buddy@gmail.com"
              rel="noreferrer"
              target="_blank"
            >
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-at flex text-xl text-black mr-2"
                data-icon="at"
                data-prefix="fas"
                focusable="false"
                role="img"
                viewBox="0 0 512 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M207.8 20.73c-93.45 18.32-168.7 93.66-187 187.1c-27.64 140.9 68.65 266.2 199.1 285.1c19.01 2.888 36.17-12.26 36.17-31.49l.0001-.6631c0-15.74-11.44-28.88-26.84-31.24c-84.35-12.98-149.2-86.13-149.2-174.2c0-102.9 88.61-185.5 193.4-175.4c91.54 8.869 158.6 91.25 158.6 183.2l0 16.16c0 22.09-17.94 40.05-40 40.05s-40.01-17.96-40.01-40.05v-120.1c0-8.847-7.161-16.02-16.01-16.02l-31.98 .0036c-7.299 0-13.2 4.992-15.12 11.68c-24.85-12.15-54.24-16.38-86.06-5.106c-38.75 13.73-68.12 48.91-73.72 89.64c-9.483 69.01 43.81 128 110.9 128c26.44 0 50.43-9.544 69.59-24.88c24 31.3 65.23 48.69 109.4 37.49C465.2 369.3 496 324.1 495.1 277.2V256.3C495.1 107.1 361.2-9.332 207.8 20.73zM239.1 304.3c-26.47 0-48-21.56-48-48.05s21.53-48.05 48-48.05s48 21.56 48 48.05S266.5 304.3 239.1 304.3z"
                  fill="currentColor"
                />
              </svg>
            </a>
          </div>
        </div>
      </div>
    </nav>
  </div>
</div>
`;
